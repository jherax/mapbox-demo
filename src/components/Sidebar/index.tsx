import styled from 'styled-components';

const Wrapper = styled.section`
  background-color: #fff;
  width: 300px;
  height: 100%;
  padding: 0px;
  box-sizing: border-box;
  border: 0 solid #e5e7eb;
`;

export default function Sidebar() {
  return <Wrapper id='Sidebar'>Sidebar</Wrapper>;
}
