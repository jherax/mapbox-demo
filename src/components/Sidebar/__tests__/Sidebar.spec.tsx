import {render, screen, within} from '@testing-library/react';

import Sidebar, {type SidebarProps} from '../index';

describe('Testing <Sidebar />', () => {
  it('should render correctly when all expected properties are present', async () => {
    const props: SidebarProps = {
      regionLogo:
        'https://images.wikia.com/halo/es/images/e/e3/BioShock_LE_Logo.png',
      legends: [
        {
          count: 3334,
          label: 'Licensed Short-Term Rental properties',
          bulletColor: '#00AAAB',
        },
      ],
      links: [
        {
          label: '24/7 hotline: ',
          showValue: '530-448-8003',
          rawValue: 'tel:530-448-8003',
        },
        {
          label: 'Complaint email: ',
          showValue: 'strcompliance@placer.ca.gov',
          rawValue: 'mailto:strcompliance@placer.ca.gov',
        },
      ],
    };

    const {asFragment} = render(<Sidebar {...props}></Sidebar>);
    const legends = getParent(screen.getByText('Program Stats'));
    const links = getParent(screen.getByText('Useful Links'));

    expect(within(legends).getAllByRole('listitem')).toHaveLength(1);
    expect(within(links).getAllByRole('listitem')).toHaveLength(2);
    expect(asFragment()).toMatchSnapshot();
  });
});

function getParent(element: HTMLElement): HTMLElement {
  return element.parentElement as HTMLElement;
}
