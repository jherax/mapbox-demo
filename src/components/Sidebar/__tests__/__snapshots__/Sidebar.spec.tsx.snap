// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Sidebar /> should render correctly when all expected properties are present 1`] = `
<DocumentFragment>
  <section
    class="sc-bdvuGq gFwqKT"
    id="Sidebar"
  >
    <header
      class="sc-gsDLol iMnAik"
    >
      <section
        id="box-titles"
      >
        <div
          id="box-image"
        >
          <img
            alt="Rentalscape"
            src="https://placer-ca-str-public-portal.deckard.com/rental-scape-logo.png"
          />
        </div>
        <div
          id="box-title"
        >
          <h1>
            Short-Term Rental (STR)
          </h1>
          <h2>
            PUBLIC PORTAL
          </h2>
        </div>
      </section>
      <section
        id="box-region-logo"
      >
        <img
          alt="region logo"
          src="https://images.wikia.com/halo/es/images/e/e3/BioShock_LE_Logo.png"
        />
      </section>
    </header>
    <article
      class="sc-dkPuFg fxVDij"
    >
      <section>
        <h2>
          Program Stats
        </h2>
        <ul>
          <li
            class="sc-hKwCKL bKMbmc"
          >
            <div>
              <span>
                3334
              </span>
              <label>
                Licensed Short-Term Rental properties
              </label>
            </div>
          </li>
        </ul>
      </section>
    </article>
    <article
      class="sc-eCImbG vjVyx"
    >
      <section>
        <h2>
          Useful Links
        </h2>
        <ul>
          <li>
            <span>
              24/7 hotline: 
            </span>
            <a
              href="tel:530-448-8003"
              rel="noopener noreferrer nofollow"
            >
              530-448-8003
            </a>
          </li>
          <li>
            <span>
              Complaint email: 
            </span>
            <a
              href="mailto:strcompliance@placer.ca.gov"
              rel="noopener noreferrer nofollow"
            >
              strcompliance@placer.ca.gov
            </a>
          </li>
        </ul>
      </section>
    </article>
    <footer
      class="sc-jRQCJB hQmAtw"
    >
      <p>
        Powered by Rentalscape
      </p>
      <p>
        Engineered by DECKARD TECHNOLOGIES
      </p>
    </footer>
  </section>
</DocumentFragment>
`;
